### DISCUSSION TITLE
Additional properties in OpenAPI specification

### DISCUSSION THREAD

**Discussion by mslonina** [directus/directus#12751] [Feature Requests]
Hi,

I'm looking into server/specs/oas and it seems that component/schemas doesn't include fields like `maximum` or `minimum` (for floats). From what I see, for each property we have:

- nullable
- description
- type

Would it be possible to include fields available in the property configuration like:

- required
- maximum
- minimum
- readOnly

This will help in creating custom frontends :)

In addition, OpenAPI specifies following in the Schema object:

- title 
- multipleOf
- exclusiveMaximum
- exclusiveMinimum
- maxLength
- minLength
- pattern 
- maxItems
- minItems
- uniqueItems
- maxProperties
- minProperties
- enum

but exposing them would probably need API support first.

**Comment by rijkvanzanten** [comment]
I'm not too familiar with the details around OAS, so I'd love a PR to get some of this going 🙂

**Comment by mslonina** [comment]
As far as I understand, it should be mapping field properties into OAS keywords. Could you point me out where to start? Readonly and required looks to be fairly straightforward. Also - there is field validation available, how it is done? Can we compute for example min/maxLength exclusiveMin/Max from the validation rules (without exposing the whole logic here)?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
